server.go

```diff
@@ -12,38 +12,44 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
```

Comment:
Could you add in `server_test.go`:
- a test with a single interceptor that tests expected behavior
- a test with two interceptors that tests ordered changes (i1, i2, final, i2, i1): the first interceptor could just add the string `i1in-`, the second could add `i2in-`, the server method handler could add `final-`, the second interceptor could add `i2out-`, and finally, the first interceptor could add `i1out`. Then the check would be that the client got the response: `i1in-i2in-final-i2out-i1out`